### PR TITLE
Add :named_level to ScriptLevel

### PR DESCRIPTION
**Pull request by joshlory**
Like 'assessment', 'named_level' specifies a different way to display a puzzle dot.  Named levels will have an icon instead of a puzzle number, show the name next to the dot, and have a line break before and after in the `Script` overview UI.

### CODE REVIEW

File: dashboard/app/models/script_level.rb [hunk 1]
```diff
@@ -2,16 +2,17 @@
...
-#  chapter    :integer
-#  created_at :datetime
-#  updated_at :datetime
-#  stage_id   :integer
-#  position   :integer
-#  assessment :boolean
-#  level_id   :integer
-#  properties :text(65535)
+#  id          :integer          not null, primary key
+#  level_id    :integer
```
Comment (mehalshah) [inline dashboard/app/models/script_level.rb:6]: Weird that this ordering change - something happen with the column orders?

Comment (joshlory) [inline dashboard/app/models/script_level.rb:6]: It was re-ordered in `schema.rb` during @balderdash's PR here: https://github.com/code-dot-org/code-dot-org/pull/8295/files#diff-11195b8114efb5813026789c8fbfbe64L559.  My guess is that `ScriptLevel` hasn't been re-annotated until now.

**Comment by mehalshah** [comment]
The change looks fine, but why are we calling these named levels? Levels already have names, is the salient different the fact that the names are displayed in the script view?

**Comment by joshlory** [comment]
Open to suggestions for a better name!

**Pull request merged**